packages/flutter/lib/src/services/scribble.dart

```diff
@@ -0,0 +1,237 @@
+  /// will break the ability to do handwriting input. This has no effect if
+  /// asserts are disabled.
+  @visibleForTesting
+  static void setChannel(MethodChannel newChannel) {
+    assert(() {
+      _instance._channel = newChannel..setMethodCallHandler(_instance._handleScribbleInvocation);
+      return true;
+    }());
+  }
+
+  static final Scribble _instance = Scribble._();
+
+  /// Set the given [ScribbleClient] as the single active client.
+  ///
+  /// This is usually based on the [ScribbleClient] receiving focus.
+  static set client(ScribbleClient? client) {
+    _instance._client = client;
+  }
+
+  /// Return the [ScribbleClient] that was most recently given focus.
+  static ScribbleClient? get client => _instance._client;
+
+  ScribbleClient? _client;
+
+  late MethodChannel _channel;
+
+  final Map<String, ScribbleClient> _scribbleClients = <String, ScribbleClient>{};
+  bool _scribbleInProgress = false;
+
+  /// Used for testing within the Flutter SDK to get the currently registered [ScribbleClient] list.
+  @visibleForTesting
+  static Map<String, ScribbleClient> get scribbleClients => Scribble._instance._scribbleClients;
+
+  /// Returns true if a scribble interaction is currently happening.
+  static bool get scribbleInProgress => _instance._scribbleInProgress;
+
+  Future<dynamic> _handleScribbleInvocation(MethodCall methodCall) async {
+    final String method = methodCall.method;
+    if (method == 'Scribble.focusElement') {
+      final List<dynamic> args = methodCall.arguments as List<dynamic>;
+      _scribbleClients[args[0]]?.onScribbleFocus(Offset((args[1] as num).toDouble(), (args[2] as num).toDouble()));
+      return;
+    } else if (method == 'Scribble.requestElementsInRect') {
+      final List<double> args = (methodCall.arguments as List<dynamic>).cast<num>().map<double>((num value) => value.toDouble()).toList();
+      return _scribbleClients.keys.where((String elementIdentifier) {
+        final Rect rect = Rect.fromLTWH(args[0], args[1], args[2], args[3]);
+        if (!(_scribbleClients[elementIdentifier]?.isInScribbleRect(rect) ?? false)) {
+          return false;
+        }
+        final Rect bounds = _scribbleClients[elementIdentifier]?.bounds ?? Rect.zero;
+        return !(bounds == Rect.zero || bounds.hasNaN || bounds.isInfinite);
+      }).map((String elementIdentifier) {
+        final Rect bounds = _scribbleClients[elementIdentifier]!.bounds;
+        return <dynamic>[elementIdentifier, ...<dynamic>[bounds.left, bounds.top, bounds.width, bounds.height]];
+      }).toList();
+    } else if (method == 'Scribble.scribbleInteractionBegan') {
+      _scribbleInProgress = true;
+      return;
+    } else if (method == 'Scribble.scribbleInteractionFinished') {
+      _scribbleInProgress = false;
+      return;
+    }
+
+    final List<dynamic> args = methodCall.arguments as List<dynamic>;
+    switch (method) {
+      case 'Scribble.showToolbar':
+        _client!.showToolbar();
```

Comment:
There's no guarantee that `_client` will be non-null I think?

Reply:
You're right.  I'm going to return before this if it's not null, so if the engine somehow sends handwriting while no field is focused, it will just be ignored.